core/src/Telemetry/DefaultEvent.cs

```diff
@@ -25,18 +25,35 @@
 //
 //------------------------------------------------------------------------------
 
-using Microsoft.Identity.Core;
+using System.Collections.Generic;
+using System.Globalization;
 
 namespace Microsoft.Identity.Core.Telemetry
 {
     internal class DefaultEvent : EventBase
     {
-        public DefaultEvent(string clientId) : base(EventNamePrefix + "default_event")
+        public DefaultEvent(string clientId, IDictionary<string, int> eventCount) : base((string) (EventNamePrefix + "default_event"))
         {
             var platformProxy = PlatformProxyFactory.GetPlatformProxy();
             this[EventNamePrefix + "client_id"] = clientId;
             this[EventNamePrefix + "sdk_platform"] = platformProxy.GetProductName()?.ToLowerInvariant();
             this[EventNamePrefix + "sdk_version"] = MsalIdHelper.GetMsalVersion();
+            this[EventNamePrefix + "application_name"] = HashPersonalIdentifier(platformProxy.GetCallingApplicationName()?.ToLowerInvariant());
+            this[EventNamePrefix + "application_version"] = HashPersonalIdentifier(platformProxy.GetCallingApplicationVersion()?.ToLowerInvariant());
+            this[EventNamePrefix + "device_id"] = HashPersonalIdentifier(platformProxy.GetDeviceId()?.ToLowerInvariant());
+            this[EventNamePrefix + "ui_event_count"] = GetEventCount(EventNamePrefix + "ui_event", eventCount);
+            this[EventNamePrefix + "http_event_count"] = GetEventCount(EventNamePrefix + "http_event", eventCount);
+            this[EventNamePrefix + "cache_event_count"] = GetEventCount(EventNamePrefix + "cache_event", eventCount);
```

Comment:
>EventNamePrefix + "cache_event_count" [](start = 17, length = 37)

seems like some of these are specified in the HttpEvent class as const's. That would be beneficial.

Reply:
None of these are specified in HttpEvent. There should be no overlap in fields for different types of events. On a separate note, it might make sense to add these all in a constants class. Adding to a work item to track.